File: ScienceClaw/backend/route/task_settings.py
from typing import Any

from fastapi import APIRouter, Depends
from pydantic import BaseModel, Field

from backend.user.dependencies import require_user, User
from backend.task_settings import (
    TaskSettings,
    UpdateTaskSettingsRequest,
    get_task_settings,
    update_task_settings,
)

router = APIRouter(prefix="/task-settings", tags=["task-settings"])


class ApiResponse(BaseModel):
    code: int = Field(default=0)
    msg: str = Field(default="ok")
    data: Any = Field(default=None)


@router.get("", response_model=ApiResponse)
async def get_settings(current_user: User = Depends(require_user)):
    settings = await get_task_settings(current_user.id)
    return ApiResponse(data=settings.model_dump())


@router.put("", response_model=ApiResponse)
async def update_settings(
    body: UpdateTaskSettingsRequest,
    current_user: User = Depends(require_user),
):
    settings = await update_task_settings(current_user.id, body)
    return ApiResponse(data=settings.model_dump())

File: ScienceClaw/frontend/src/api/taskSettings.ts
import { apiClient, type ApiResponse } from './client';

export interface TaskSettings {
  agent_stream_timeout: number;
  sandbox_exec_timeout: number;
  max_tokens: number;
  output_reserve: number;
  max_history_rounds: number;
  max_output_chars: number;
}

export type UpdateTaskSettingsRequest = Partial<TaskSettings>;

export async function getTaskSettings(): Promise<TaskSettings> {
  const response = await apiClient.get<ApiResponse<TaskSettings>>('/task-settings');
  return response.data.data;
}

export async function updateTaskSettings(data: UpdateTaskSettingsRequest): Promise<TaskSettings> {
  const response = await apiClient.put<ApiResponse<TaskSettings>>('/task-settings', data);
  return response.data.data;
}
